apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1.IntegrationTests/DatastoreTransactionTest.cs

```diff
@@ -150,5 +150,31 @@ private async Task CommitTest(Func<DatastoreTransaction, Task<CommitResponse>> c
             var descriptions = entities.Select(e => (string) e["description"]);
             Assert.Equal(new[] { "Inserted in transaction", "Upserted in transaction", "Updated in transaction" }, descriptions);
         }
+
+        [Fact]
+        public void Transaction_WithCount()
+        {
+            var db = _fixture.CreateDatastoreDb();
+            var keyFactory = db.CreateKeyFactory("CountTestTransaction");
+            var entities = new[]
+            {
+                new Entity { Key = keyFactory.CreateKey("x"), ["description"] = "description for x" },
+                new Entity { Key = keyFactory.CreateKey("y"), ["description"] = "description for y" }
+            };
+            db.Insert(entities);
+            using var transaction = db.BeginTransaction();
+            var gqlQuery = new GqlQuery { QueryString = "SELECT count(*)  as `count` FROM CountTestTransaction " };
+            var query = new Query("CountTestTransaction");
+            AggregationQuery aggQuery = new AggregationQuery(query)
+            {
+                Aggregations = { Aggregations.Count("count") }
+            };
+            AggregationQueryResults resultsForStructuredQuery = transaction.RunAggregationQuery(aggQuery);
+            AggregationQueryResults resultsForGqlQuery = transaction.RunAggregationQuery(gqlQuery);
+            long count1 = resultsForStructuredQuery["count"].IntegerValue;
+            long count2 = resultsForGqlQuery["count"].IntegerValue;
+            Assert.Equal(2, count2);

```

Comment:
Nit: reverse order of these assertions, just to keep things ordered?

Or possibly inline each:

```csharp
Assert.Equal(2, resultsForStructuredQuery["count"].IntegerValue);
Assert.Equal(2, resultsForGqlQuery["count"].IntegerValue);
```